### ISSUE TITLE
Cannot resize/create raw disk image

### ISSUE THREAD

**Issue by loganwilliams** [docker/for-mac#2383]
### Expected behavior
* Docker launches

### Actual behavior
* Docker does not launch: com.docker.supervisor failed to start. Exit code 1

### Information

Resetting settings to default does not fix this issue. Removing all Docker folders, as below, does not fix this issue.

```
sudo rm -r /var/tmp/com.docker.vmnetd.socket
sudo rm -r ~/Library/Application\ Scripts/com.docker.helper/
sudo rm -r ~/Library/Containers/com.docker.*
sudo rm -r ~/Library//Preferences/com.docker.docker.plist
sudo rm -r ~/Library/Group\ Containers/group.com.docker/
```

Issue #2317 appears similar, but I do not have any extra folders in the filesharingDirectories list, and altering this list does not change the behavior.

```
Docker for Mac: version: 17.12.0-ce-mac46 (a61e84b8bca06b1ae6ce058cdd7beab1520ad622)
macOS: version 10.13.2 (build: 17C88)
logs: /tmp/279C93EC-A70B-46F1-9C0F-F937ED7BBCFA/20180109-151953.tar.gz
failure: com.docker.vpnkit is not running
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[ERROR]  driver.amd64-linux
         com.docker.driver.amd64-linux is not running
[OK]     virtualization VT-X
[OK]     app
[ERROR]  moby
         /Users/loganw/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/console-ring does not exist
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     env
[OK]     virtualization kern.hv_support
[ERROR]  slirp
         Unexpected error ((Failure
  "Error connecting socket to 9p endpoint unix:/Users/loganw/Library/Containers/com.docker.docker/Data/s51: Unix.Unix_error(Unix.ECONNREFUSED, \"connect\", \"\")")) connecting to /Users/loganw/Library/Containers/com.docker.docker/Data/s51
         com.docker.vpnkit is not running
         slirp check failed with: Unix.Unix_error(Unix.ECONNREFUSED, "connect", "")
[ERROR]  osxfs
         com.docker.osxfs is not running
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         Connection refused (ECONNREFUSED) connecting to /var/run/docker.sock: check if service is running
         Connection refused (ECONNREFUSED) connecting to /Users/loganw/Library/Containers/com.docker.docker/Data/s60: check if service is running
         docker ps failed
[OK]     menubar
[OK]     disk
```

### Steps to reproduce the behavior

  1. Download Docker version: 17.12.0-ce-mac46
  2. Run Docker.app

**Comment by vsipuli** [comment]
Seeing the same, Docker gets stuck in "Docker is starting". Tried reinstalling, but to no help. Haven't found any way to get it running, so cannot use Docker anymore.

**Comment by mamartins** [comment]
Same problem here!

**Comment by erel-levine** [comment]
same problem, different build.
I too do not have any extra folders in the filesharingDirectories list, and neither "reset to factory" nor rebooting the mac change anything. 
so no docker for now. sad!

Docker for Mac: version: 17.12.0-ce-mac46 (a61e84b8bca06b1ae6ce058cdd7beab1520ad622)
macOS: version 10.13.2 (build: 17C205)
logs: /tmp/90B2AAA8-3C33-429E-BB95-2BFAA373870D/20180110-164641.tar.gz
failure: com.docker.vpnkit is not running
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[ERROR]  driver.amd64-linux
         com.docker.driver.amd64-linux is not running
[OK]     virtualization VT-X
[OK]     app
[ERROR]  moby
         /Users/erellevine/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/console-ring does not exist
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     env
[OK]     virtualization kern.hv_support
[ERROR]  slirp
         Unexpected error ((Failure
  "Error connecting socket to 9p endpoint unix:/Users/erellevine/Library/Containers/com.docker.docker/Data/s51: Unix.Unix_error(Unix.ECONNREFUSED, \"connect\", \"\")")) connecting to /Users/erellevine/Library/Containers/com.docker.docker/Data/s51
         com.docker.vpnkit is not running
         slirp check failed with: Unix.Unix_error(Unix.ECONNREFUSED, "connect", "")
[ERROR]  osxfs
         com.docker.osxfs is not running
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         Connection refused (ECONNREFUSED) connecting to /var/run/docker.sock: check if service is running
         Connection refused (ECONNREFUSED) connecting to /Users/erellevine/Library/Containers/com.docker.docker/Data/s60: check if service is running
         docker ps failed
[OK]     menubar
[OK]     disk

**Comment by mamartins** [comment]
Temporary fix: Install the edge version

**Comment by loganwilliams** [comment]
@mamartins The edge version actually has the same issue for me.

**Comment by tonyta** [comment]
I'm seeing the same problem on the latest Stable and Edge builds.

```
Docker for Mac: version: 17.12.0-ce-mac45 (a61e84b8bca06b1ae6ce058cdd7beab1520ad622)
macOS: version 10.13.2 (build: 17C205)
logs: /tmp/D4280AA8-4858-4BB0-A965-77035CFEB2CA/20180111-115733.tar.gz
failure: com.docker.vpnkit is not running
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[ERROR]  driver.amd64-linux
         com.docker.driver.amd64-linux is not running
[OK]     virtualization VT-X
[OK]     app
[ERROR]  moby
         /Users/tonyta/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/console-ring does not exist
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     env
[OK]     virtualization kern.hv_support
[ERROR]  slirp
         Unexpected error ((Failure
  "Error connecting socket to 9p endpoint unix:/Users/tonyta/Library/Containers/com.docker.docker/Data/s51: Unix.Unix_error(Unix.ECONNREFUSED, \"connect\", \"\")")) connecting to /Users/tonyta/Library/Containers/com.docker.docker/Data/s51
         com.docker.vpnkit is not running
         slirp check failed with: Unix.Unix_error(Unix.ECONNREFUSED, "connect", "")
[ERROR]  osxfs
         com.docker.osxfs is not running
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         Connection refused (ECONNREFUSED) connecting to /var/run/docker.sock: check if service is running
         Connection refused (ECONNREFUSED) connecting to /Users/tonyta/Library/Containers/com.docker.docker/Data/s60: check if service is running
         docker ps failed
[OK]     menubar
[OK]     disk
```

**Comment by mamartins** [comment]
Sorry to ear that, fortunately for me edge version fixed it.

Just to be sure you did the same as me: I've moved the docker.app to transh and then ruined the uninstall.sh provided on their site. Then I rebooted and then I installed edge version.

Good luck guys!

**Comment by tonyta** [comment]
Thanks, @mamartins. Rebooting after initial install allowed the Docker client to get past whatever the problem was. I was able to get both the latest Edge (`17.12.0-ce-mac45`) and Stable (`17.12.0-ce-mac46`) versions to run.

**Comment by souljacker** [comment]
+1 
same issue

**Comment by toribr** [comment]
+ 1 same issue

**Comment by akimd** [comment]
Hi!

In your logs (docker-system-os.log), we can find 
```
2018-01-09 15:19:48.764027-0500  localhost com.docker.driver.amd64-linux[1685]: Cannot create/resize "/Users/loganw/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.raw": truncate /Users/loganw/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.raw: no space left on device status 1
```

The name of the file (*.raw) shows that Docker for Mac decided to use the Raw format, rather than Qcow.  It does that when the disk image is stored on an SSD with APFS.  This is indeed the case, right?

What does `df -h ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux` gives for you?

Also, could you please make sure you have the GNU Coreutils?  Say via brew or Mac Ports.  We want to play with `truncate` (called `gtruncate` on MacPorts).

```
$ gtruncate --version
truncate (GNU coreutils) 8.29
Copyright © 2017 Free Software Foundation, Inc.
License GPLv3+ : GNU GPL version 3 ou ultérieure <https://www.gnu.org/licenses/gpl.fr.html>
Ceci est un logiciel libre. Vous êtes libre de le modifier et de le redistribuer.
Ce logiciel n'est accompagné d'ABSOLUMENT AUCUNE GARANTIE, dans les limites
permises par la loi.
Écrit par Padraig Brady.
```

Could you please try this:

```
$ gtruncate -s 64GiB ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
$ ls -ls ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
0 -rw-r--r--  1 akim  staff  68719476736 16 jan 10:58 /Users/akim/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
$ du -h ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
  0B    /Users/akim/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
```

As you can see, it creates a file whose capacity is 64GiB, but since it's empty, it consumes only 0B.  Your error seems to indicate that the file failed to be sparse.

**Comment by loganwilliams** [comment]
Thank you @akimd. Here is the result of those commands.

```
diverging-windmill:~ loganw$  df -h ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux
Filesystem     Size   Used  Avail Capacity iused               ifree %iused  Mounted on
/dev/disk1s1  234Gi  219Gi   11Gi    96% 3516394 9223372036851259413    0%   /
diverging-windmill:~ loganw$ gtruncate -s 64GiB ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
gtruncate: failed to truncate '/Users/loganw/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw' at 68719476736 bytes: No space left on device
diverging-windmill:~ loganw$ ls -ls ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
0 -rw-r--r--  1 loganw  staff  0 Jan 16 09:09 /Users/loganw/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
diverging-windmill:~ loganw$ du -h ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
  0B	/Users/loganw/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
diverging-windmill:~ loganw$ 
```

**Comment by akimd** [comment]
This is interesting!  You are short in space, and apparently spare files are not supported.

Could you run the Mac's system report (from "About this mac") and check the storage section?  I have this (excuse my French):

<img width="862" alt="capture d ecran 2018-01-16 a 18 53 01" src="https://user-images.githubusercontent.com/232441/35003889-8c05b5a4-faee-11e7-9cae-f9f05b364f60.png">

I am particularly interested in the filesystem (APFS for me).

**Comment by loganwilliams** [comment]
It looks like this (yes, I have APFS.) Docker worked fine on this machine last time I tried, which was before upgrading to APFS.

<img width="863" alt="screen shot 2018-01-16 at 12 57 15 pm" src="https://user-images.githubusercontent.com/823027/35004122-f5760dae-fabc-11e7-9729-3567313e7ab6.png">

**Comment by akimd** [comment]
Bummer.

We have to understand why everything seems ok for sparse files support, yet it fails.

It used to work before because it was simply not used, that's a new feature (faster, etc.).

**Comment by loganwilliams** [comment]
Interestingly, making an 11 GiB file with truncate, and then expanding it to 64 GiB works without an error.

```
diverging-windmill:~ loganw$ rm ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
diverging-windmill:~ loganw$ gtruncate -s 11GiB ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
diverging-windmill:~ loganw$ gtruncate -s 64GiB ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
diverging-windmill:~ loganw$ ls -ls ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
0 -rw-r--r--  1 loganw  staff  68719476736 Jan 16 13:31 /Users/loganw/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
```

**Comment by akimd** [comment]
Wow...  Maybe the OS tries to keep some headroom and does it badly?

It's hard to understand the behaviour here, but I can reproduce something similar.
```
$ df -h /tmp/
Filesystem     Size   Used  Avail Capacity iused               ifree %iused  Mounted on
/dev/disk1s1  466Gi  426Gi   34Gi    93% 3175399 9223372036851600408    0%   /
$ rm -f /tmp/big && gtruncate -s 1TiB /tmp/big
gtruncate: impossible de tronquer '/tmp/big' à 1099511627776 octets: No space left on device
$ rm -f /tmp/big && gtruncate -s 500GiB /tmp/big
gtruncate: impossible de tronquer '/tmp/big' à 536870912000 octets: No space left on device
$ rm -f /tmp/big && gtruncate -s 200GiB /tmp/big
gtruncate: impossible de tronquer '/tmp/big' à 214748364800 octets: No space left on device
$ rm -f /tmp/big && gtruncate -s 100GiB /tmp/big
$ gtruncate -s 1TiB /tmp/big
gtruncate: impossible de tronquer '/tmp/big' à 1099511627776 octets: No space left on device
$ gtruncate -s 500GiB /tmp/big
gtruncate: impossible de tronquer '/tmp/big' à 536870912000 octets: No space left on device
$ gtruncate -s 200GiB /tmp/big
$ gtruncate -s 400GiB /tmp/big
gtruncate: impossible de tronquer '/tmp/big' à 429496729600 octets: No space left on device
$ df -h /tmp
Filesystem     Size   Used  Avail Capacity iused               ifree %iused  Mounted on
/dev/disk1s1  466Gi  426Gi   34Gi    93% 3175399 9223372036851600408    0%   /
```
Maybe @djs55 will have some ideas.  It's funny indeed that we can start small and grow later.  It's like the frog in the heated water.

The behaviour is really peculiar...
```
$ gtruncate -s 400GiB /tmp/big
gtruncate: impossible de tronquer '/tmp/big' à 429496729600 octets: No space left on device
$ ls -ls /tmp/big
0 -rw-r--r--  1 akim  wheel  214748364800 16 jan 20:19 /tmp/big
$ gtruncate -s 300GiB /tmp/big
$ gtruncate -s 330GiB /tmp/big
$ gtruncate -s 360GiB /tmp/big
$ gtruncate -s 390GiB /tmp/big
$ gtruncate -s 399GiB /tmp/big
$ gtruncate -s 400GiB /tmp/big
$ gtruncate -s 500GiB /tmp/big
$ gtruncate -s 800GiB /tmp/big
gtruncate: impossible de tronquer '/tmp/big' à 858993459200 octets: No space left on device
$ gtruncate -s 600GiB /tmp/big
$ gtruncate -s 700GiB /tmp/big
$ gtruncate -s 800GiB /tmp/big
$ gtruncate -s 900GiB /tmp/big
$ gtruncate -s 1TiB /tmp/big
$ ls -ls /tmp/big
0 -rw-r--r--  1 akim  wheel  1099511627776 16 jan 20:21 /tmp/big
$ ls -lsh /tmp/big
0 -rw-r--r--  1 akim  wheel   1,0T 16 jan 20:21 /tmp/big
```

**Comment by liujj** [comment]
Also reported a same issue on my MBP, previously started up but cracked after an update.

Tried everything I can find but not work for me.

Logs in docker-system-os.log are below:
```
2018-01-16 17:05:03.083305+0800  localhost com.docker.driver.amd64-linux[25015]: Acquired hypervisor lock
2018-01-16 17:05:03.084078+0800  localhost com.docker.driver.amd64-linux[25015]: TRIM is enabled; recycling thread will keep 262144 sectors free and will compact after 262144 more sectors are free
2018-01-16 17:05:03.084156+0800  localhost com.docker.driver.amd64-linux[25015]: Attempting to resize "/Users/liujj/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.raw" from 0MiB to 65536MiB
2018-01-16 17:05:03.084221+0800  localhost com.docker.driver.amd64-linux[25015]: Cannot create/resize "/Users/liujj/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.raw": truncate /Users/liujj/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.raw: no space left on device
2018-01-16 17:05:03.090791+0800  localhost com.docker.supervisor[25012]: Reap com.docker.driver.amd64-linux (pid 25015): exit status 1
```

So I tried cmds above:

```
$ df -h ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux
Filesystem     Size   Used  Avail Capacity iused               ifree %iused  Mounted on
/dev/disk1s1  186Gi  178Gi  6.0Gi    97% 2401811 9223372036852373996    0%   /

$ gtruncate -s 64GiB ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
gtruncate: failed to truncate '/Users/liujj/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw' at 68719476736 bytes: No space left on device

$ ls -ls ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
0 -rw-r--r--  1 liujj  staff  0 Jan 17 11:25 /Users/liujj/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
```

When I try to run a command with 11GiB file, it works (my disk space available is 6.0GiB). And then it failed when I try expanding it to 64 GiB.

```
$ rm ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw

$ gtruncate -s 11GiB ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw

$ gtruncate -s 64GiB ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
gtruncate: failed to truncate '/Users/liujj/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw' at 68719476736 bytes: No space left on device

$ ls -ls ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
0 -rw-r--r--  1 liujj  staff  11811160064 Jan 17 11:28 /Users/liujj/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/foo.raw
```

Is there anyway to customize this value at current stable or edge version. Or any method to let the docker run in temporary

<img width="778" alt="wx20180117-113714 2x" src="https://user-images.githubusercontent.com/3626540/35024511-ca0c227a-fb7a-11e7-92d3-4da8a0d9abf0.png">

**Comment by akimd** [comment]
@liujj Yes, you have a similar pattern: when you are short on space, there's a limit on the growth of a sparse file.  I venture that you went too fast from 11GiB to 64GiB: making smaller increments would probably succeed.

**Comment by akimd** [comment]
@liujj and @loganwilliams Could you please try the following script?
```
#! /bin/sh

set -e

to_bytes ()
{
    local v=$1
    case $v in
        (*KiB) to_bytes "$((${v/KiB} * 1024))";;
        (*MiB) to_bytes "$((${v/MiB} * 1024))KiB";;
        (*GiB) to_bytes "$((${v/GiB} * 1024))MiB";;
        (*TiB) to_bytes "$((${v/TiB} * 1024))GiB";;

        (*KB) to_bytes "$((${v/KB} * 1000))";;
        (*MB) to_bytes "$((${v/MB} * 1000))KB";;
        (*GB) to_bytes "$((${v/GB} * 1000))MB";;
        (*TB) to_bytes "$((${v/TB} * 1000))GB";;

        (*) echo "$1";
    esac
}

dimg=$(sed -En 's/.*diskPath.*:.*"(.*)".*/\1/p' < "$HOME/Library/Group Containers/group.com.docker/settings.json")

size=$(to_bytes "${1-64GB}")
raw=${2-$dimg}
step=$(to_bytes "${3-1GiB}")

echo "Building $raw of $size bytes by increments of $step"
s=0
while ((s < size)); do
    ((s += step ))
    if (( size < s )); then
        (( s=size ))
    fi
    echo "Grow to $s"
    dd if=/dev/zero of="$raw" bs=1 count=0 seek="$s"
done
echo "Built $raw of $size bytes"
```
Save it somewhere as `create-raw`, run `chmod +x create-raw`, and then `./create-raw <SIZE> <PATH> <STEP>`.  By default SIZE is 64GB, PATH is the disk image, and STEP is 1GB.

It should create the disk image, and you should be able to start Docker.

Beware that it will destroy your disk image if you have one and care about it.  Please report success/failure.

**Comment by mamartins** [comment]
In my case I have +- 110Gb of free space when this happened to me

**Comment by akimd** [comment]
@mamartins Can you still reproduce that problem?  Starting from a clear state.  Also, could you try to see, with the commands above, if you can create a large sparse file by hand?

**Comment by akimd** [comment]
Could you please try this modified version of Docker 17.12?  Hopefully it should fix your issue.  If it does, we will soon release an official stable image which includes the fix.  Once you tested it, be sure to reinstall a "real" stable release.

https://download-stage.docker.com/mac/pr/21923/Docker.dmg

**Comment by loganwilliams** [comment]
The modified version of Docker did not appear to change anything. After running the script that you  provided however, I was able to successfully launch the Docker stable channel.

**Comment by akimd** [comment]
@loganwilliams When you tried the version I provided, did you install it and let it performs a "reset to factory defaults"?  Could you please submit again diagnostics once that very Docker failed?

**Comment by loganwilliams** [comment]
Hm, maybe I was mistaken. I just tried to reproduce it and it successfully launched. I ran the same set of rm commands that I listed in my first post prior to launching it. Sorry about that.

**Comment by akimd** [comment]
Good news, thanks!

**Comment by loganwilliams** [comment]
I just managed to reproduce it by:
* Running your script and installing a stable Docker version (successfully launched)
* Deleting Docker.app and copying the modified version you provided to /Applications
* Running the modified Docker.app and letting it reset to factory default settings (when I ran the rm commands first, it did not ask to do this, presumably because there was no trace of Docker on my system, so this must be the difference.)
* It then fails to launch with seemingly the same symptoms as my original report.

The diagnostic ID is 88D92910-DC38-466D-A3C3-62FFC910C46F.

**Comment by gtardif** [comment]
@loganwilliams Thanks for retrying and sending the diags, I might see what is not working in this test build, different codepaths depending on whether the raw file already exists or not (event sized to zero). 
If you could you try to : 
- quit without uninstalling the test build
- manually rm the Docker.raw file, and then recreate one with size zero
(`gtruncate -s 0 ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.raw`) 
- restart Docker.app, hopefully this will resize properly the raw file to 64G and Docker should work fine. Let us know how it goes (you can also re-upload diags).
In the meantime I'll work on another test build that handles this properly.

**Comment by gtardif** [comment]
When you get a chance, you can completely uninstall, then try the test build https://download-stage.docker.com/mac/pr/21935/Docker.dmg. Hopefully this one should work with or without an existing Docker.raw file.

**Comment by toribr** [comment]
I resolved the disk space issue (spotlight was incorrectly reporting) on my computer and now Docker is installing correctly.

**Comment by akimd** [comment]
Could you please check if the following image fixes your problems? You might need to reset to factory defaults...

https://download-stage.docker.com/mac/pr/21949/Docker.dmg

This is a test build. Ignore any update it might propose, and after testing, reinstall an Edge or Stable version to get proper updates again. Thanks!

**Comment by liujj** [comment]
The script `create-raw` successfully created a 64GiB image, and the test build from [pr21949](https://download-stage.docker.com/mac/pr/21949/Docker.dmg) works well on my MBP.
Then I profromed an update on edge channel, all goes well and thanks very much.

**Comment by jefyian** [comment]
solved by :
**chmod 600 /Library/LaunchDaemons/com.docker.vmnetd.plist
sudo launchctl load -Fw /Library/LaunchDaemons/com.docker.vmnetd.plist**

from:https://thecodecentral.com/

**Comment by akimd** [comment]
@jefyian I don't think it has anything to go with it.

**Comment by jefyian** [comment]
@akimd Maybe it just worked for me.

**Comment by loganwilliams** [comment]
@akimd The latest version works great for me. I was unable to reproduce the error following the steps in my post on January 17. Sorry for the delay in testing this out.

**Comment by gtardif** [comment]
We just released 17.12.0-ce-mac49 on the Stable channel that contains the fix mentioned above. (and 18.01 on the Edge channel also fixes this)
